src/main/java/com/vfasad/service/ReportService.java

```diff
@@ -0,0 +1,232 @@
+                    .stream()
+                    .mapToDouble( (userByTeam) -> getUserBonus(userByTeam, options, areaByTeam))
+                    .sum();
+            reportMonthTeam.setTeamName(team.getName());
+            reportMonthTeam.setArea(areaByTeam);
+            reportMonthTeam.setIncome(calculateSumPrice(ordersByTeam));
+            reportMonthTeam.setPaintExpenses(calculateSumPaintExpenses(ordersByTeam));
+            reportMonthTeam.setPaintPrice(calculateSumPriceConsumes(ordersByTeam));
+            reportMonthTeam.setSalary(salary);
+            reportMonthTeam.setBonus(bonus);
+            reportMonthTeam.setDoneBy(doneBy);
+            reportMonthTeamRepository.save(reportMonthTeam);
+        });
+    }
+
+    private void saveReportMonthEmployee(int year, int month, Map<String, Option> options, User doneBy) {
+        List<Order> orders = orderService.findByMonth(year, month);
+        Map<User, List<Order>> ordersByUserId = new HashMap<>();
+        for (Order order : orders) {
+            for (User user : order.getDoneBy()) {
+                ordersByUserId.putIfAbsent(user, new ArrayList<>());
+                ordersByUserId.get(user).add(order);
+            }
+        }
+        ordersByUserId.forEach((user, ordersByUser) -> {
+            ReportMonthEmployee reportMonthEmployee = new ReportMonthEmployee(year, month, user.getId());
+            reportMonthEmployee.setEmployeeName(user.getName());
+            double sumArea = calculateSumArea(ordersByUserId.get(user), options);
+            reportMonthEmployee.setArea(sumArea);
+            reportMonthEmployee.setSalary(getUserSalary(user, options));
+            reportMonthEmployee.setBonus(getUserBonus(user, options, sumArea));
+            reportMonthEmployee.setRole(user.getRole());
+            reportMonthEmployee.setDoneBy(doneBy);
+            reportMonthEmployeeRepository.save(reportMonthEmployee);
+        });
+    }
+
+    private double calculateSumArea(List<Order> orders, Map<String, Option> options) {
+        double clipToArea = parseDouble(options.get(OptionName.CLIP_TO_AREA.toString()).getValue());
+        double furnitureSmallToArea = parseDouble(options.get(OptionName.FURNITURE_SMALL_TO_AREA.toString()).getValue());
+        double furnitureBigToArea = parseDouble(options.get(OptionName.FURNITURE_BIG_TO_AREA.toString()).getValue());
+        return orders.stream()
+                .mapToDouble((o) -> o.getArea() +
+                        o.getClipCount() * clipToArea +
+                        o.getFurnitureSmallCount() * furnitureSmallToArea +
+                        o.getFurnitureBigCount() * furnitureBigToArea)
+                .sum();
+    }
+
+    private double calculateElectricity(double sumArea, double electricity) {
+        return sumArea * electricity;
+    }
+
+    private double calculateSumPriceConsumes(List<Order> orders) {
+        return orders.stream()
+                .mapToDouble((o) ->
```

Comment:
It's legal to avoid braces around single argument in lambda, so let's avoid useless braces to increase readability